django/db/models/sql/query.py

```diff
@@ -151,6 +155,9 @@ def _execute_query(self):
         self.cursor = connection.cursor()
         self.cursor.execute(self.sql, params)
 
+    def resolve_expression(self, query, *args, **kwargs):
```

Comment:
The `resolve_expression` method of the _resolvable_ interface must return a _compilable_ which is an object with an `as_sql` method as mentioned in the ticket.

If the test currently pass it must be because `__iter__` is implicitly called by `In(Lookup)` and you are not actually executing the `RawQuery`'s SQL as a subquery like it should be.

What I believe should happen here is for `resolve_expression` to return copy of `self` that has a flag set to denote that it is used as a subquery `e.g. `clone = self.clone(self.using); clone.subquery = True` and an `as_sql` method that return something like `f"({self.sql})" if self.subquery else self.sql, params` which the params tweaks of `_execute_query`.

From there `_execute_query` could be adapted to call `self.as_sql` to pass it to the cursor